### ISSUE TITLE
DSSP Support for Python 3.9+

### ISSUE THREAD

**Issue by a-r-j** [salilab/conda-recipes#16]
Hiya, it seems newer versions of python are not currently supported.

```bash
conda create -n dssp python=3.9
conda activate dssp
conda install -c salilab dssp
```

Output:
```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: |
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - dssp -> python[version='2.7.*|3.5.*|3.6.*|3.4.*']

Your python: python=3.9

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.
```

**Comment by benmwebb** [comment]
Right, we built this package years ago so while it doesn't depend on Python, one of the dependencies at the time (most likely Boost) does, and that old version of Boost was not rebuilt for newer Pythons. Probably we could rebuild DSSP against a newer Boost if you're using Linux or Windows (it won't build on Mac, issue #15). But I would recommend you use the newest DSSP - this is a very old package, not maintained by the original authors any more.

**Comment by a-r-j** [comment]
Thanks for getting back to me!

I see, thanks for the context. I assume the newest DSSP would be distributed by `apt-get install dssp`?

I think a rebuild would reasonably useful; conda indicates this version has been downloaded 25k times so I think there's moderate demand & it would make packaging it as a dependency easier.

**Comment by benmwebb** [comment]
> I assume the newest DSSP would be distributed by `apt-get install dssp`?

I'm not 100% sure as I haven't used DSSP for years, but yes, it looks like there is a DSSP .deb package, at least when I looked on Ubuntu 22.04.

> I think a rebuild would reasonably useful

I was able to build a package against the latest Boost on Linux at least - that should work with newer Pythons.

**Comment by a-r-j** [comment]
Thanks again for this. I've found that the latest DSSP (`3.0.0`) won't install without `libboost-1.7.3` (t[he conda version was recently updated to `1.8.2`](https://anaconda.org/anaconda/libboost)).

**Comment by adalal78** [comment]
@a-r-j , I'm recently trying to install dssp using the command conda install salilab::dssp, and get the following error:
Channels:
 - defaults
 - salilab
Platform: win-64
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - salilab::dssp

Current channels:

  - defaults
  - https://conda.anaconda.org/salilab
 

Did you get this or something else?  Anyone can help me out?

**Comment by benmwebb** [comment]
> @a-r-j , I'm recently trying to install dssp using the command conda install salilab::dssp, and get the following error: 

@adalal78 The dssp package we built is only available for Linux and Mac: https://anaconda.org/salilab/dssp

Looks like you are trying to install on Windows, so that won't work; just like `conda` says, the package is not available.

This is unrelated to @a-r-j's issue trying to install with newer Pythons.

**Comment by eunos-1128** [comment]
@a-r-j 

See https://github.com/salilab/conda-recipes/issues/19#issuecomment-3677509426 if you still need to use dssp v3.
